packages/docs/docs/docs/linking.md

```diff
@@ -1,127 +1,281 @@
 
-contract MyLinkedContract {
-  ERC721Mintable private _token;
+contract TokenExchange is Initializable {
+  using SafeMath for uint256;
 
-  function setToken(ERC721Mintable token) external {
-    require(address(token) != address(0));
-    _token = token;
+  // Contract state: exchange rate and token
+  uint256 public rate;
+  IERC20 public token;
+
+  // Initializer function (replaces constructor)
+  function initialize(uint256 _rate, IERC20 _token) public initializer {
+    rate = _rate;
+    token = _token;
   }
-}
+
+  // Send tokens back to the sender using predefined exchange rate
+  function() external payable {
+    uint256 tokens = msg.value.mul(rate);
+    token.transfer(msg.sender, tokens);
+  }
+} 
 ```
 
-One thing to notice is that instead of importing `openzeppelin-solidity` we are
-importing `openzeppelin-eth`. This is the name of the OpenZeppelin EVM package,
-the one we have to use if we want to reuse the package already deployed.
-For more information, see
-[this article which explains the difference between openzeppelin-solidity and openzeppelin-eth](https://blog.zeppelin.solutions/getting-started-with-openzeppelin-eth-a-new-stable-and-upgradeable-evm-package-576fb37297d0#125e).
+Note the usage of the `initializer` modifier in the `initialize` method. This guarantees that, after we have deployed our contract, no one can call into that function again to alter the token or the rate.
 
-Now, let's link our project to the openzeppelin-eth package by running:
+Let's now create and initialize our new `TokenExchange` contract:
 
 ```console
-npx zos link openzeppelin-eth
+$ zos create
+✓ Compiled contracts with solc 0.5.9 (commit.e560f70d)
+? Pick a contract to instantiate: TokenExchange
+? Pick a network: development
+✓ Contract TokenExchange deployed
+? Do you want to call a function on the instance after creating it?: Yes
+? Select which function: initialize(_rate: uint256, _token: address)
+? _rate (uint256): 10
+? _token (address): 0x2612Af3A521c2df9EAF28422Ca335b04AdF3ac66
+Instance created at 0x26b4AFb60d6C903165150C6F0AA14F8016bE4aec
 ```
 
-This command will install the openzeppelin-eth contracts locally, which is
-needed to compile the contracts that import it; but it will also update the
-`zos.json` file with a reference to the linked package to make sure that when
-you deploy the project it will use the EVM package that already exists on the
-blockchain.
+<!-- CODE: We need to extend the zos transfer command to support erc20s -->
 
-The following commands will be familiar to you. Just as we did before, we have
-to add the contract to the project:
+Our exchange is almost ready! We only need to fund it, so it can send tokens to purchasers. Let's do that using the `send-tx` command, to transfer the full token balance from our own account to the exchange contract. Make sure to replace the recipient of the transfer with the `TokenExchange` address you got from the previous command.
```

Comment:
I had to enter two inputs, A rate, and the contract address. but it didn't talk about it. 

we have this line above "Our exchange contract will need to store the token contract address and the exchange rate in its state." 

we can change it to 

Our TokenExchange contract will need to store the token contract address and the exchange rate in its state. We will provide these values at the time of contract deployment.